group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -8570,6 +8645,8 @@ private boolean maybeDeleteEmptyStreamsGroup(String groupId, List<CoordinatorRec
             // Add tombstones for the previous streams group. The tombstones won't actually be
             // replayed because its coordinator result has a non-null appendFuture.
             createGroupTombstoneRecords(group, records);
+            // Cancel initial rebalance timer.
+            timer.cancel(streamsInitialRebalanceKey(groupId));
```

Comment:
I think we have to move this into `createGroupTombstoneRecords`, so that it's also done when delete group is called